libs/@local/hql/cst/src/value.rs

```diff
@@ -0,0 +1,46 @@
+use alloc::borrow::Cow;
+
+use hql_span::SpanId;
+use json_number::Number;
+
+use crate::{arena, Spanned};
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct Entry<'arena, 'source> {
```

Comment:
(discussed offline) To be renamed, maybe `ObjectEntry`

Reply:
I really disliked the design of this, so I've gone ahead and did a bit of redesign in [a83f408](https://github.com/hashintel/hash/pull/4828/commits/a83f4082ecdb45d8cce8ecfbd40f65b28206ae4d), the span is now stored on the key instead, but indexing is still done with a str.